service/history/shardContext.go

```diff
@@ -86,6 +86,7 @@ type (
 		GetCurrentTime(cluster string) time.Time
 		GetTimerMaxReadLevel(cluster string) time.Time
 		UpdateTimerMaxReadLevel(cluster string) time.Time
+		GetEncoding(domainEntry *cache.DomainCacheEntry) common.EncodingType
```

Comment:
We should try and keep History Engine logic completely agnostic of Encoding or any serialization/deserialization info.  It should work on strongly typed objects and serialization is taken care of at persistence layer.

Reply:
Make sense will be done as part of https://github.com/uber/cadence/issues/1718